src/store/student-store.js

```diff
@@ -153,6 +154,12 @@ store.registerHandler('CHECKIN', data => {
         return;
     }
     let student = store.getStudentById(data.studentId);
+    if(store.hasOverdueItem(data.studentId)){
```

Comment:
NACK: Is this the right metric for a "problem student"? Unless I misunderstand, this would penalize a student for returning items that are not overdue (if they have other items that are overdue). It would also penalize a student for returning items in two sequential checkins more than a student returning the exact same items in one checkin.